src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs

```diff
@@ -382,6 +375,15 @@ private string GetRelativePath(RouteTemplate parsedTemplate)
                 }
             }
 
+            // Set the default status only when no status has already been set explicitly
+            if (objectTypes.Count == 0)
+            {
+                if (type != null && type != typeof(void))
+                {
+                    objectTypes[StatusCodes.Status200OK] = type;
+                }
+            }
```

Comment:
Since this all seems contingent on the `objectTypes` what will happen by default if the method returns `void`